packages/expo-modules-core/android/src/main/java/expo/modules/interfaces/taskManager/TaskServiceProviderHelper.kt

```diff
@@ -0,0 +1,36 @@
+package expo.modules.interfaces.taskManager
+
+import android.content.Context
+import expo.modules.core.ModulePriorities
+import expo.modules.core.interfaces.Package
+
+class TaskServiceProviderHelper {
+  companion object {
+    /*
+      Uses reflection to look through the current list of packages and attempts to find
+      one that provides a TaskServiceInterface implementation.
+     */
+    fun getTaskServiceImpl(context: Context): TaskServiceInterface? {
```

Comment:
would `ApplicationContext` (or whatever it's actually called) be more appropriate here, to avoid the `ClassCastException` that we saw yesterday?

Reply:
TaskService expects a Context - and the application context is a ReactApplicationContext so I thought to try to follow the TaskService and type it as simple as possible. I can add a comment about this - that makes a lot of sense.

Reply:
Added a comment in the function doc about this - hopefully that should be enough?